Plugin/Model/Order/ItemRepositoryPlugin.php

```diff
@@ -0,0 +1,154 @@
+     * @param GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode
+     * @param SourceSelectionServiceInterface $sourceSelectionService
+     */
+    public function __construct(
+        OrderItemExtensionFactory $orderItemExtensionFactory,
+        GetSourceSelectionResultFromOrder $sourceSelectionResult,
+        GetInventoryRequestFromOrder $getInventoryRequestFromOrder,
+        GetDefaultSourceSelectionAlgorithmCodeInterface $getDefaultSourceSelectionAlgorithmCode,
+        SourceSelectionServiceInterface $sourceSelectionService
+    ) {
+        $this->orderItemExtensionFactory = $orderItemExtensionFactory;
+        $this->sourceSelectionResult = $sourceSelectionResult;
+        $this->getInventoryRequestFromOrder = $getInventoryRequestFromOrder;
+        $this->getDefaultSourceSelectionAlgorithmCode = $getDefaultSourceSelectionAlgorithmCode;
+        $this->sourceSelectionService = $sourceSelectionService;
+    }
+
+    /**
+     * Set additional attributes when requesting a single order item
+     *
+     * @param ItemRepository  $subject
+     * @param OrderItemInterface|Item $result
+     * @return OrderItemInterface|Item
+     */
+    public function afterGet(ItemRepository $subject, OrderItemInterface $result)
+    {
+        return $this->applyExtensionAttributesToOrderItem($result, $result);
+    }
+
+    /**
+     * Set additional attributes when requesting a list of order items
+     *
+     * @param ItemRepository $subject
+     * @param OrderItemSearchResultInterface $result
+     * @return OrderItemSearchResultInterface
+     */
+    public function afterGetList(ItemRepository $subject, OrderItemSearchResultInterface $result)
+    {
+        foreach ($result->getItems() as $item) {
+            $this->applyExtensionAttributesToOrderItem($item, $result->getItems());
+        }
+
+        return $result;
+    }
+
+    /**
+     * Apply extension attributes to order item
+     *
+     * @param OrderItemInterface $orderItem
+     * @param array $allItems
+     * @return OrderItemInterface
+     */
+    private function applyExtensionAttributesToOrderItem(OrderItemInterface $orderItem, array $allItems): OrderItemInterface
+    {
+        if (!$extensionAttributes = $orderItem->getExtensionAttributes()) {
+            $extensionAttributes = $this->orderItemExtensionFactory->create();
+        }
+
+        $extensionAttributes->setSourceCode(
+            $this->getOrderSourceCode($orderItem->getOrder(), $allItems)
+        );
+
+        $orderItem->setExtensionAttributes($extensionAttributes);
+
+        return $orderItem;
+    }
+
+    /**
+     * @TODO what happens if there's more than one source?
+     *
+     * @param Order $order
+     * @param array $allItems
+     * @return string|null
+     */
+    private function getOrderSourceCode(Order $order, array $allItems): ?string
+    {
+        if (empty($this->sourceSelectionResultValues)) {
```

Comment:
It might be a better idea to cache this by order ID. What would happen if this repository functionality was called multiple times with different order items with different order ID's? How about having an array of `$sourceCodeByOrderId` indexed by order ID, for example:
```php
<?php

class ItemRepositoryPlugin
{
    private $sourceCodeByOrderId = [];

    ...

    private function getOrderSourceCode(Order $order, array $allItems): ?string
    {
        $orderId = $order->getId();
        if (array_key_exists($orderId, $this->sourceCodeByOrderId)) {
            return $this->sourceCodeByOrderId[$orderId];
        }

        ...

        $sourceSelectionItems = $sourceSelectionResult->getSourceSelectionItems();
        return $this->sourceCodeByOrderId[$orderId] = !empty($sourceSelectionItems) ? $sourceSelectionItems[0]->getSourceCode() : null;
    }
}
```